presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -700,6 +702,14 @@ default void revokeTablePrivileges(ConnectorSession session, SchemaTableName tab
         throw new PrestoException(NOT_SUPPORTED, "This connector does not support page sink commit");
     }
 
+    /**
+     * Handles metadata update requests and sends the results back to worker
+     */
+    default List<MetadataUpdateResult> updateMetadata(List<MetadataUpdateRequest> metadataUpdateRequests, String queryId)
```

Comment:
```java
ConnectorMetadataUpdateHandle updateMetadata(ConnectorSession session, ConnectorMetadataUpdateHandle handle)
```

Likely you can squeeze input and output into one object. But if not, feel free to create two handle interfaces